### PR TITLE
feat: Update to more recent python

### LINKED ISSUE #1: Add support for python 3.9, 3.10, 3.11, 3.12 & drop 2.7
Hey there.
I really appreciate this repo and learned a lot about ndef looking at the code and the examples.

I'd like to propose a shift of supported python versions, as a few years have passed.
Mind if I opened a PR to do so?

### PR DESCRIPTION
**Pull request by AiyionPrime**
This one resolves #1 and contains minor necessary changes to the example which would be invalid without.
Commits are all [conventional](https://www.conventionalcommits.org/en/v1.0.0/), this branch could cleanly be merged by rebasing.

Tests are not done yet, just wanted to give you a draft before concluding this later tonight.

### CODE REVIEW

File: examples/create.py [hunk 1]
```diff
@@ -3,9 +3,9 @@
...
 smart_poster_message = ndef.new_smart_poster('Github', 'http://github.com/')
 smart_poster_raw_ndef = smart_poster_message.to_buffer()
-print 'smart poster ndef:', smart_poster_raw_ndef.encode('hex')
+print('smart poster ndef:', smart_poster_raw_ndef.encode('hex'))
 
 text_record = (ndef.TNF_WELL_KNOWN, ndef.RTD_TEXT, 'id', 'hello world')
 text_message = ndef.new_message(text_record)
 text_raw_ndef = text_message.to_buffer()
-print 'text ndef:', text_raw_ndef.encode('hex')
+print('text ndef:', text_raw_ndef.encode('hex'))
```
Comment (kichik) [inline examples/create.py]: This should now be `text_raw_ndef.hex()`, no?

**Comment by kichik** [comment]
@AiyionPrime is this ready to merge or were you planning on adding more?

**Comment by AiyionPrime** [comment]
This looks great; one sec :)

**Comment by AiyionPrime** [comment]
No, this looks great.
I've got a bunch of typehints lying around, but that'll be a different PR.
Thanks for saving me the headache around six!

**Pull request merged**